src/ansi-c/goto-conversion/link_to_library.cpp

```diff
@@ -41,26 +41,32 @@ add_one_function(
   // convert to CFG
   if(
     library_model.symbol_table.symbols.find(missing_function) !=
-    library_model.symbol_table.symbols.end())
+      library_model.symbol_table.symbols.end() &&
+    library_model.symbol_table.lookup_ref(missing_function).value.is_not_nil())
   {
     goto_convert(
       missing_function,
       library_model.symbol_table,
       library_model.goto_functions,
       message_handler);
+    library_model.symbol_table.get_writeable_ref(missing_function)
+      .set_compiled();
```

Comment:
Would it be reasonable to ask to add a comment explaining the impact of setting this as compiled? In this PR is very clear, but it might not be just looking at the code.

Reply:
Done!